doc/source/cluster/cluster_under_construction/ray-clusters-on-kubernetes/user-guides/logging.md

```diff
@@ -2,9 +2,149 @@
+:::{tip}
+Skip to {ref}`the deployment instructions<kuberay-logging-tldr>`
+for a sample configuration showing how to extract logs from a Ray pod.
 :::
 
-Notes on logging, including instructions for setting up fluentd.
-A guide to logging using fluentd.
+## The Ray log directory
+Each Ray pod runs several component processes, such as the Raylet, object manager, dashboard agent, etc.
+These components log to files in the directory `/tmp/ray/session_latest/logs` in the pod's file system.
+Extracting and persisting these logs requires some setup.
+
+## Log processing tools
+There are a number of log processing tools available within the Kubernetes
+ecosystem. This page will shows how to extract Ray logs using [Fluent Bit][FluentBit].
+Other popular tools include [Fluentd][Fluentd], [Filebeat][Filebeat], and [Promtail][Promtail].
+
+## Log collection strategies
+We mention two strategies for collecting logs written to a pod's filesystem,
+**sidecar containers** and **daemonsets**. You can read more about these logging
+patterns in the [Kubernetes documentation][KubDoc].
+
+### Sidecar containers
+We will provide an {ref}`example<kuberay-fluentbit>` of the sidecar strategy in this guide.
+You can process logs by configuring a log-processing sidecar
+for each Ray pod. Ray containers should be configured to share the `/tmp/ray`
+directory with the logging sidecar via a volume mount.
+
+You can configure the sidecar to do either of the following:
+* Stream Ray logs to the sidecar's stdout.
+* Export logs to an external service.
+
+### Daemonset
+Alternatively, it is possible to collect logs at the Kubernetes node level.
+To do this, one deploys a log-processing daemonset onto the Kubernetes cluster's
+nodes. With this strategy, it is key to mount
+the Ray container's `/tmp/ray` directory to the relevant `hostPath`.
+
+(kuberay-fluentbit)=
+# Setting up logging sidecars with Fluent Bit.
+In this section, we give an example of how to set up log-emitting
+[Fluent Bit][FluentBit] sidecars for Ray pods.
+
+## Configure log processing
+The first step is to create a ConfigMap with configuration
+for FluentBit.
+
+Here is a minimal ConfigMap which tells a Fluent Bit sidecar to
+* Tail Ray logs.
+* Output the logs to the container's stdout.
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: fluentbit-config
+data:
+  fluent-bit.conf: |
+    [INPUT]
+        Name tail
+        Path /tmp/ray/session_latest/logs/*
+        Tag ray
+        Path_Key true
+        Refresh_Interval 5
+    [OUTPUT]
+        Name stdout
+        Match *
+```
+In addition to streaming logs to stdout, you can export logs to any
+[storage backend][FluentBitStorage] supported by Fluent Bit.
+
+## Add logging sidecars to your RayCluster CR.
+
+### Add log and config volumes.
+For each pod template in our RayCluster CR, we
+need to add two volumes: One volume for Ray's logs
+and another volume to store Fluent Bit configuration from the ConfigMap
+applied above.
+```yaml
+volumes:
```

Comment:
This will be much more user friendly if we show the user how to integrate it into their configuration. We could have a link like

https://github.com/ray-project/ray/blob/2cd1b7bc9d8b726c2f4b9dabe18cecd6cf2a8a8f/doc/source/cluster/cluster_under_construction/ray-clusters-on-kubernetes/configs/ray-cluster.log.yaml#L86

included here or maybe sphinx has some way to do this :)

Same for the other two snippets below!

Reply:
Makes sense, let me think about how organize these.

Reply:
Add a link, embedded the yaml in the page, and made the snippets subsets of the embedded yaml.